packages/twenty-front/src/modules/object-record/object-filter-dropdown/components/ObjectFilterDropdownSourceSelect.tsx

```diff
@@ -61,6 +62,7 @@ export const ObjectFilterDropdownSourceSelect = ({
   const { currentViewWithCombinedFiltersAndSorts } =
     useGetCurrentView(viewComponentId);
 
+  // TODO: this should be removed as it is not consistent across re-renders
```

Comment:
@lucasbordeau there is a hook in React that you can use: useId() that should be consistent accross re-renders

Reply:
useId() is not consistent across mount/unmount and as been designed to help with SSR more than providing reliable ids across time.

Reply:
I think that we should rely on fieldMetadataId + filterGroupId, which is what I'm trying to do right now to anticipate advanced filters refactor.